presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftColumnData.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.client;
+
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import javax.annotation.Nullable;
+
+import static com.facebook.swift.codec.ThriftField.Requiredness.OPTIONAL;
+import static java.util.Objects.requireNonNull;
+
+@ThriftStruct
+public final class ThriftColumnData
+{
+    private final boolean[] nulls;
```

Comment:
I think it's one of the questions that should be more clear once I adda readme for the connector and its supposed usage. Thrift interface (client) can and will be implemented in languages other than java. C++ is the what upcoming users will use. As such, this interface, among other things, needs to be relatively easy to implement and should be independent of block serialization implementation, unless it's straightforward or it's strictly necessary.
Suggested interface is one of the ways to have a compact data representation which satisfies those points.